OutOfSchool/OutOfSchool.DataAccess/Repository/ProviderRepository.cs

```diff
@@ -58,14 +58,13 @@ public new async Task Delete(Provider entity)
         {
             db.Entry(entity).State = EntityState.Deleted;
 
-            if (entity.LegalAddressId == entity.ActualAddressId)
-            {
-                db.Entry(new Address { Id = entity.LegalAddressId }).State = EntityState.Deleted;
-            }
-            else
+            // TODO: Q: why we are using new entity instead of Provider.LegalAddress here and belowe ?
+            db.Entry(new Address { Id = entity.LegalAddressId }).State = EntityState.Deleted;
+
+            if (entity.ActualAddressId.HasValue
+                && entity.ActualAddressId.Value == entity.LegalAddressId)
```

Comment:
Change `==` to `!=`
As we need to delete ActualAddress if it is not equal to Legal